streams/src/test/java/org/apache/kafka/streams/integration/EosBetaUpgradeIntegrationTest.java

```diff
@@ -147,9 +154,16 @@
     private final AtomicInteger commitCounterClient2 = new AtomicInteger(-1);
     private final AtomicInteger commitRequested = new AtomicInteger(0);
 
-    private Throwable uncaughtException;
-
     private int testNumber = 0;
+    private boolean hasUnexpectedError = false;
```

Comment:
This variable should be declare `volatile` because the callback is executed on a different thread.

Reply:
Nice catch! Fixed.